perfact/zodbsync/commands/watch.py

```diff
@@ -509,6 +520,10 @@ def run(self, interval=10):
         else:
             self.spawned_setup()
         while not self.exit.is_set():
-            self.step()
+            try:
+                self.step()
+            except TreeOutdatedException:
+                self.release_lock()
```

Comment:
Why is this needed?
EDIT: OK, I see now that it is needed because `step` explicitly uses `acquire_lock` and `release_lock`, not taking care of releasing it in case of an exception.

Could you please change it such that `step` either uses `with` or `try ... finally`? That function should be responsible for releasing the lock, not `run` - otherwise each test that calls `step` manually will need to remember to release the lock. In particular, `run` is untested (and should probably be marked as such - I was first wondering why this PR reduced the test coverage, but this is the reason) and should therefore contain as few logic as possible (which is why I split everything essential into the `step` function when I finally managed to add tests for the watcher functionality).